SplunkRumWorkspace/SplunkRum/SplunkRum/ConnectionInstrumentation.swift

```diff
@@ -0,0 +1,181 @@
+            let keyStr = key as? String
+            if keyStr != nil {
+                if keyStr?.caseInsensitiveCompare("server-timing") == .orderedSame {
+                    let valStr = val as? String
+                    if valStr != nil {
+                        if valStr!.starts(with: "traceparent") {
+                            addLinkToSpan(span: span, valStr: valStr!)
+                        }
+                    }
+                }
+            }
+        }
+    }
+    if error != nil {
+        span.setAttribute(key: "error", value: true)
+        span.setAttribute(key: "exception.message", value: error!.localizedDescription)
+        span.setAttribute(key: "exception.type", value: String(describing: type(of: error!)))
+    }
+    span.setAttribute(key: "http.response_content_length_uncompressed",
+                      value: Int(hresponse?.expectedContentLength ?? 0))
+    if hostConnectionType != nil {
+        span.setAttribute(key: "net.host.connection.type", value: hostConnectionType!)
+    }
+    span.end()
+}
+func swizzleClassMethod(clazz: AnyClass, orig: Selector, swizzled: Selector) {
+    let origM = class_getClassMethod(clazz, orig)
+    let swizM = class_getClassMethod(clazz, swizzled)
+    if origM != nil && swizM != nil {
+        method_exchangeImplementations(origM!, swizM!)
+    } else {
+        debug_log("warning: could not swizzle "+NSStringFromSelector(orig))
+    }
+}
+func initalizeConnectionInstrumentation() {
+    let connection = NSURLConnection.self
+    let swizSyncSel = #selector(NSURLConnection.splunk_swizzled_connection_sendSynchronousRequest(_:returning:))
+    let asySel =  #selector((NSURLConnection.splunk_swizzled_sendAsynchronousRequest(request:queue:completionHandler:)))
+
+    swizzleClassMethod(clazz: connection,
+                       orig: #selector(NSURLConnection.sendAsynchronousRequest(_:queue:completionHandler:)),
+                       swizzled: asySel)
+    swizzleClassMethod(clazz: connection,
+                       orig: #selector(NSURLConnection.sendSynchronousRequest(_:returning:)),
+                       swizzled: swizSyncSel)
+}
+
+/**
+  Extension of NSURLConnection class for swizzled methods.
+ */
+extension NSURLConnection {
+    /**
+     Adds a callback whenever the sendSynchronousRequest called.
+     */
+    @objc open class func
+    splunk_swizzled_connection_sendSynchronousRequest
+    (_ request: URLRequest?, returning response: AutoreleasingUnsafeMutablePointer<URLResponse?> )
+    throws -> Data {
+        let span = startConnectionSpan(request: request)
+        var data = Data()
+        do {
+            data = try splunk_swizzled_connection_sendSynchronousRequest(request, returning: response)
+            if let httpResponse = response.pointee as? HTTPURLResponse {
+                endConnectionSpan(connection: nil, hresponse: httpResponse, error: nil, span: span!)
+            }
+        } catch let error {
```

Comment:
Won't we break users' code if we don't rethrow?

Reply:
throw error